### PR TITLE
RHCLOUD-25597 | feature: use different metrics for the failures

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
This will allow us to understand which errors we are suffering from in this particular integration.

## Links
[[RHCLOUD-25597]](https://issues.redhat.com/browse/RHCLOUD-25597)

### CODE REVIEW

**Comment by MikelAlejoBR** [comment]
There is one really weird thing, though: I am unable to clear the counters' values after/before the tests, and the increments are read wrong because of that :thinking:

**Comment by gwenneg** [comment]
@MikelAlejoBR Could you please rebase this PR?

**Comment by MikelAlejoBR** [comment]
Looking into the failures...

**Comment by MikelAlejoBR** [comment]
There is something about the counter: I think the tests are saving the "export.service.failureserror..." counter, but when counting the increments, it seems that `getOrDefault` is just returning zero, because it doesn't find the counter, I am assuming? And therefore the thing doesn't work:

debug

**Review by gwenneg — COMMENTED** [review]
This may help fix the issues you had.

File: engine/src/test/java/com/redhat/cloud/notifications/exports/ExportEventListenerTest.java [hunk 1]
```diff
@@ -155,7 +219,12 @@ void testNonParseableEventsToFilterRaisesError() {
 
         // Assert that the errors counter was incremented, and that the
         // successes counter did not increment.
-        this.micrometerAssertionHelper.assertCounterIncrement(ExportEventListener.EXPORTS_SERVICE_FAILURES_COUNTER, 1);
+        this.micrometerAssertionHelper.assertCounterIncrement(
+            ExportEventListener.EXPORTS_SERVICE_FAILURES_COUNTER,
+            1,
+            ExportEventListener.FAILURE_KEY,
+            ExportEventListener.FAILURE_UNABLE_EXTRACT_FILTERS
+        );
```
[the remark this answers is not in the document]
Comment (MikelAlejoBR) [reply engine/src/test/java/com/redhat/cloud/notifications/exports/ExportEventListenerTest.java]: Ooof! It was right there :facepalm: ... That worked, thank you very much!

**Review by gwenneg — APPROVED** [review]
LGTM

**Pull request merged**